cmds/core/dmesg/dmesg_test.go

```diff
@@ -80,14 +74,23 @@ func TestDmesg(t *testing.T) {
 					t.Errorf("dmesg() = '%v', want: '%v'", err, tt.err)
 				}
 			} else {
-				if tt.buf.String() != "A" && *clear {
+				if tt.buf.String() != "A" && contains(tt.args, "-c") {
 					t.Errorf("System log should be cleared")
-				} else if !strings.Contains(tt.buf.String(), buf.String()) && *readClear {
+				} else if !strings.Contains(tt.buf.String(), buf.String()) && contains(tt.args, "-r") {
 					t.Errorf("System log should contain %s", buf.String())
-				} else if tt.buf.String() == "" && (!*clear && !*readClear) {
+				} else if tt.buf.String() == "" && (!contains(tt.args, "-c") && !contains(tt.args, "-r")) {
 					t.Errorf("System log should not be cleared")
 				}
 			}
 		})
 	}
 }
+
+func contains(arr []string, val string) bool {
```

Comment:
consider using 
https://pkg.go.dev/slices#Contains

does this same thing

Reply:
:100: